core/src/main/java/com/linecorp/armeria/internal/common/CancellationScheduler.java

```diff
@@ -26,10 +26,16 @@
 public interface CancellationScheduler {
 
     static CancellationScheduler ofClient(long timeoutNanos) {
+        if (timeoutNanos == 0) {
+            timeoutNanos = Long.MAX_VALUE;
+        }
         return new DefaultCancellationScheduler(timeoutNanos, false);
     }
 
     static CancellationScheduler ofServer(long timeoutNanos) {
+        if (timeoutNanos == 0) {
+            timeoutNanos = Long.MAX_VALUE;
+        }
         return new DefaultCancellationScheduler(timeoutNanos, true);
     }
```

Comment:
What should we do when a user specified `Long.MAX_VALUE`? Should we use `Long.MAX_VALUE - 1` if we handle `Long.MAX_VALUE` specially? :thinking:

Reply:
Yeah, but correctness matters :rofl: